### ISSUE TITLE
Rule Change: no-case-declarations: Option to ignore when not used in uninitialized state

### ISSUE THREAD

**Issue by NightProgramming** [eslint/eslint#15851]
### What rule do you want to change?

no-case-declarations

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
// Examples of incorrect code for this rule

switch (foo) {
    case 1:
        let x = 1;
        break;    
}
```

### What does the rule currently do for this code?

Gives "Unexpected lexical declaration in case block."

### What will the rule do after it's changed?

No warning/error because `x` is not used while having an uninitialized state.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

Could there be an option for this rule so that no warning/error is shown when the declared part (`x` in this sample) is not used outside of the `case` where it is declared? Because if not used in other `case`s/`default` then it cannot be unitialized.

**Comment by btmills** [comment]
I could _maybe_ see adding an "unless referenced only with in a single `case`" mode to this rule. The downside there is future changes could cause non-local errors. Generally, we try to build rules such that writing code can only introduce errors in that code. It can be unexpected to change code in one place and get an error in another. "If I'm declaring in a `case`, wrap it in curlies" is a much simpler rule for developers to keep in mind.

**Comment by nzakas** [comment]
Closing as we won’t be making a change here.
